src/common/format/helpers.js

```diff
@@ -1,8 +1,17 @@
 import { createSlug } from 'speakingurl';
 
-export const lowercase = (s) => s.toLowerCase();
-export const shellquote = (s) =>
+export const lowercase = () => (s) => s.toLowerCase();
```

Comment:
Is this what smart people refer to as "currying"? Compiling functions from functions?

Reply:
Not strictly, but you've got the right idea of how this works. 👍 

It's not _currying_ in the strict sense, because if you look at its definition, it's "[a translation] of a function that takes multiple arguments into evaluating a sequence of functions, each with a single argument". — https://en.wikipedia.org/wiki/Currying

Our new helper "creators" here can take more than a single argument in the first step (see `substring = (...args) => …`).

So really what we've done is turn the helper functions, and changed them from `String -> String` to `[Any] -> String -> String` (roughly) to allow configuration of helpers in the first evaluation step.